WooCommerce/Classes/POS/Presentation/PointOfSaleEntryPointView.swift

```diff
@@ -44,12 +45,13 @@ struct PointOfSaleEntryPointView: View {
                                  totalsViewModel: totalsViewModel,
                                  cartViewModel: cartViewModel,
                                  itemListViewModel: itemListViewModel)
-            .onAppear {
-                onPointOfSaleModeActiveStateChange(true)
-            }
-            .onDisappear {
-                onPointOfSaleModeActiveStateChange(false)
-            }
+        .environmentObject(posModalManager)
```

Comment:
This looks like the perfect case to use `.environmentObject()`, so we do not need to pass the object explicitly to all the views that might need a modal 💯

Reply:
I'd still like to neaten it up more, perhaps it's possible to do something in the `posRootModal` modifier so we don't have to remember to do both. 

It's really sensitive to where you do it as well - I can't inject it in the Dashboard, regardless of modifier order, it has to be a level up from that (which might prevent us doing the above.)